nav2_collision_monitor/src/velocity_polygon.cpp

```diff
@@ -0,0 +1,239 @@
+        "/set_polygon"));
+    polygon_sub_topic_ =
+      node->get_parameter(
+      polygon_name_ + "." + velocity_polygon_name_ +
+      ".polygon_sub_topic").as_string();
+
+    if (!polygon_sub_topic_.empty()) {
+      RCLCPP_INFO(
+        logger_,
+        "[%s][%s]: Subscribing on %s topic for polygon",
+        polygon_name_.c_str(), velocity_polygon_name_.c_str(), polygon_sub_topic_.c_str());
+      rclcpp::QoS polygon_qos = rclcpp::SystemDefaultsQoS();  // set to default
+      polygon_sub_ = node->create_subscription<geometry_msgs::msg::PolygonStamped>(
+        polygon_sub_topic_, polygon_qos,
+        std::bind(&VelocityPolygon::polygonCallback, this, std::placeholders::_1));
+    }
+
+    // linear_max param
+    nav2_util::declare_parameter_if_not_declared(
+      node, polygon_name_ + "." + velocity_polygon_name_ + ".linear_max",
+      rclcpp::ParameterValue(0.0));
+    linear_max_ =
+      node->get_parameter(
+      polygon_name_ + "." + velocity_polygon_name_ +
+      ".linear_max").as_double();
+
+    // linear_min param
+    nav2_util::declare_parameter_if_not_declared(
+      node, polygon_name_ + "." + velocity_polygon_name_ + ".linear_min",
+      rclcpp::ParameterValue(0.0));
+    linear_min_ =
+      node->get_parameter(
+      polygon_name_ + "." + velocity_polygon_name_ +
+      ".linear_min").as_double();
+
+    // direction_end_angle param
+    nav2_util::declare_parameter_if_not_declared(
+      node, polygon_name_ + "." + velocity_polygon_name_ + ".direction_end_angle",
+      rclcpp::ParameterValue(0.0));
+    direction_end_angle_ =
+      node->get_parameter(
+      polygon_name_ + "." + velocity_polygon_name_ +
+      ".direction_end_angle").as_double();
+
+    // direction_start_angle param
+    nav2_util::declare_parameter_if_not_declared(
+      node, polygon_name_ + "." + velocity_polygon_name_ + ".direction_start_angle",
+      rclcpp::ParameterValue(0.0));
+    direction_start_angle_ =
+      node->get_parameter(
+      polygon_name_ + "." + velocity_polygon_name_ +
+      ".direction_start_angle").as_double();
+
+    // theta_max param
+    nav2_util::declare_parameter_if_not_declared(
+      node, polygon_name_ + "." + velocity_polygon_name_ + ".theta_max", rclcpp::ParameterValue(
+        0.0));
+    theta_max_ =
+      node->get_parameter(polygon_name_ + "." + velocity_polygon_name_ + ".theta_max").as_double();
+
+    // theta_min param
+    nav2_util::declare_parameter_if_not_declared(
+      node, polygon_name_ + "." + velocity_polygon_name_ + ".theta_min", rclcpp::ParameterValue(
+        0.0));
+    theta_min_ =
+      node->get_parameter(polygon_name_ + "." + velocity_polygon_name_ + ".theta_min").as_double();
+
+  } catch (const std::exception & ex) {
+    RCLCPP_ERROR(
+      logger_,
+      "[%s][%s]: Error while getting polygon parameters: %s",
+      polygon_name_.c_str(), velocity_polygon_name_.c_str(), ex.what());
+    return false;
```

Comment:
Nothing of substance happens here, you simply drop the error on the floor and continue as if nothing happened!

Reply:
fixed with proper return of failure case